src/llmcompressor/transformers/sparsification/compressed_tensors_utils.py

```diff
@@ -77,15 +81,22 @@ def save_pretrained_wrapper(
             if state_dict is None:
                 state_dict = get_state_dict_offloaded_model(model)
 
+            sparsity_stucture = SparsityConfigMetadata.infer_sparsity_structure(model)
+            quantization_format = infer_quantization_format(
+                model=model,
+                quantization_format=quantization_format,
+                save_compressed=save_compressed,
+                sparsity_structure=sparsity_stucture,
+            )
+            is_marlin = quantization_format == CompressionFormat.marlin_24.value
+
             if sparsity_config is not None:
                 sparsity_config.global_sparsity = (
```

Comment:
tbh we should just get rid of this branch and instead add `SparsityConfigMetadata.from_format`, but that's out of scope

Reply:
Since really the only point in being able to pass sparsity config is to be able to override the format afaict

Reply:
I agree; we'll make those updates soon!